File: back/src/main/java/com/back/domain/review/dto/AdminReviewDto.java
package com.back.domain.review.dto;

import com.back.domain.member.dto.MemberDto;
import com.back.domain.review.entity.Review;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDateTime;
import java.util.List;

public record AdminReviewDto(
        @NotNull
        Long id,
        @NotNull
        String bookTitle,
        @NotNull
        Float rating,
        @NotNull
        String content,
        @NotNull
        LocalDateTime createdDate,
        @NotNull
        MemberDto reviewer,
        @NotNull
        List<String> tags
) {
    public AdminReviewDto(Review review) {
        this(
                review.getId(),
                review.getBook().getTitle(),
                review.getRating(),
                review.getContent(),
                review.getCreatedDate(),
                new MemberDto(review.getReviewer()),
                review.getTags());
    }
}

File: back/src/main/java/com/back/domain/review/dto/AdminReviewDto.kt
package com.back.domain.review.dto

import com.back.domain.member.dto.MemberDto
import com.back.domain.review.entity.Review
import java.time.LocalDateTime

class AdminReviewDto(
    val id: Long,
    val bookTitle: String,
    val rating: Float,
    val content: String,
    val createdDate: LocalDateTime,
    val reviewer: MemberDto,
    val tags: List<String>
) {
    constructor(review: Review) : this(
        review.id,
        review.book.title,
        review.rating,
        review.content,
        review.createdDate,
        MemberDto(review.reviewer),
        review.tags
    )
}